packages/gatsby-cli/src/reporter/reporter-phantom.ts

```diff
@@ -0,0 +1,36 @@
+import reporterActions from "./redux/actions"
+import { ActivityStatuses, ActivityTypes } from "./constants"
+import { Span } from "opentracing"
+
+export interface ICreatePhantomReporterArgumnents {
+  text: string
+  id: string
+  span: Span
+}
+
+export type PhantomReporter = ReturnType<typeof createPhantomReporter>
+
+export const createPhantomReporter = ({
+  text,
+  id,
+  span,
+}: ICreatePhantomReporterArgumnents) => {
+  return {
+    start: () => {
+      reporterActions.startActivity({
+        id,
+        text,
+        type: ActivityTypes.Hidden,
+      })
+    },
+    end() {
```

Comment:
Any reason why this is mixing arrow functions and functions? These need return types too, or a linter exception and explanation.